### DISCUSSION TITLE
Interfacing with Several GPIO Pins

### DISCUSSION THREAD

**Discussion by VR314** [nasa/fprime#2349] [Q&A] [answered]
We are trying to write a component that interfaces with 10-20 GPIO pins. My interpretation of the current LinuxGPIODriver implementation is that you can only set one pin for each instance of the LinuxGPIODriver in the topology, and you also cannot update the pin number dynamically. How could we get around this, besides instantiating a LinuxGPIODriver instance for every pin we may need to interface with?

**Comment by ebenezerarunkumar** [comment]
I believe you do have to create a new instance for every possible GPIO pin you are working with! At least that’s what I had to do in the past…

**Comment by timcanham** [comment, marked as answer]
@VR314 The LinuxGPIODriver is an example of a driver that implements a single GPIO, but there isn't any reason you can't spin your own that implements the behavior you want. It might make the best sense for your use case.
